src/Modules/Authontication/Components/Register/Register.tsx

```diff
@@ -1,9 +1,271 @@
-import React from 'react'
-
+import { useState } from 'react'
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify'
+import { USERS_URLs } from '../../../../constans/END_POINTS';
+import regImg from '../../../../assets/images/regImg.png'
+import { EmailValidation, FieldValidation, PasswordValidation } from '../../../../constans/VALIDATIONS';
+import { useForm } from 'react-hook-form';
+import axios from 'axios';
 
 export default function Register() {
+      
+let navigate= useNavigate();
+let{
+  register,
+  handleSubmit,
+  getValues,
+  formState:{errors},
+} = useForm();
+
+const convertFormdataToObj = (data:any) => {
+  const formdata = new FormData()
+  formdata.append("userName", data.userName)
+  formdata.append("country", data.country)
+  formdata.append("password", data.password)
+  formdata.append("email", data.email)
+  formdata.append("phoneNumber", data.phoneNumber)
+  formdata.append("confirmPassword", data.confirmPassword)
+  data?.profileImage && formdata.append('profileImage', data.profileImage[0])
+  return formdata
+}
+
+const onSubmit = async (data:any) => {
+  const registerData = convertFormdataToObj(data)
+  try {
+    const response = await axios.post(USERS_URLs.Register, registerData)
+
+    console.log(response);
+    navigate("/verify-account")
+    toast.success(response?.data.message)
+  }
+  catch (error:any) {
+    toast.error(
+     error?.response?.data?.message || "Register unsuccessful. Please try again"
+   );
+    console.log(error);
```

Comment:
We should display the error message coming from the backend and if there is no one, we can provide a default message.